core/cas-server-core-services-registry/src/main/java/org/apereo/cas/services/AbstractServicesManager.java

```diff
@@ -104,10 +96,17 @@ public RegisteredService findServiceBy(final String serviceId) {
             return null;
         }
 
-        val service = getCandidateServicesToMatch(serviceId)
-            .filter(r -> r.matches(serviceId))
-            .findFirst()
-            .orElse(null);
+        var service = getCandidateServicesToMatch(serviceId)
+                .filter(r -> r.matches(serviceId))
+                .findFirst()
+                .orElse(null);
+
+        if (service == null) {
+            service = serviceRegistry.findServiceBy(serviceId);
+            if (service != null) {
+                services.put(service.getId(), service);
+            }
+        }
```

Comment:
It would be good to add some log statements here under trace to explain the behavior and flow.